src/overlay_01_02204ED8.c

```diff
@@ -0,0 +1,215 @@
+#include "assert.h"
+#include "encounter.h"
+#include "field_follow_poke.h"
+#include "field_map_object.h"
+#include "fieldmap.h"
+#include "filesystem.h"
+#include "heap.h"
+#include "overlay_01.h"
+#include "overlay_01_021EDAFC.h"
+#include "overlay_02.h"
+#include "player_data.h"
+#include "scrcmd.h"
+#include "script.h"
+#include "task.h"
+#include "unk_0205CB48.h"
+#include "unk_0205FD20.h"
+
+typedef struct UnkStruct_02204EFC {
+    u16 unk00;
+    u16 unk02;
+    void *unk04;
+    u8 filler08[20];
+    void *unk1c;
+    u8 filler20[20];
+    void *unk34;
+    u8 filler38[20];
+    s16 unk4c[][2];
+} UnkStruct_02204EFC;
+
+static BOOL ov01_02204EFC(TaskManager*);
+static s8 ov01_02204FE0(u16, u16, u32, u32, u32, s16[][2]);
+static s8 ov01_02205074(u8, u8, u32);
+static void ov01_022050F8(FieldSystem*, u32*, u32*);
+
+const s8 ov01_02209634[] = {
+    0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 0,
+};
+
+const s8 ov01_02209640[][2] = {
+    {0, 1},
+    {1, 0},
+    {0, 1},
+    {1, 0},
+    {0, 1},
+    {1, 0},
+    {0, 1},
+    {1, 0},
+    {0, 1},
+    {1, 0},
+};
+
+const s8 ov01_02209654[][3] = {
+    { 0,  1, -1},
+    { 0, -1,  1},
+    { 1,  0, -1},
+    { 1, -1,  0},
+    {-1,  0,  1},
+    {-1,  1,  0},
+    { 0,  1, -1},
+    { 0, -1,  1},
+    { 1,  0, -1},
+    { 1, -1,  0},
+};
+
+const s8 ov01_02209672[][4] = {
+    {-1,  0,  0,  1},
+    { 1, -1,  0,  0},
+    { 1,  1, -1,  0},
+    { 0,  1,  1, -1},
+    { 0,  0,  1,  1},
+    {-1,  0,  1,  0},
+    { 1, -1,  0,  1},
+    { 0,  1, -1,  0},
+    { 1,  0,  1, -1},
+    { 0,  1,  0,  1},
+};
+
+const s8 ov01_0220969A[][5] = {
+    {-1,  0,  0,  1,  1},
+    { 1, -1,  0,  0,  1},
+    { 1,  1, -1,  0,  0},
+    { 0,  1,  1, -1,  0},
+    { 0,  0,  1,  1, -1},
+    {-1,  0,  1,  0,  1},
+    { 1, -1,  0,  1,  0},
+    { 0,  1, -1,  0,  1},
+    { 1,  0,  1, -1,  0},
+    { 0,  1,  0,  1, -1},
+};
+
+void ov01_02204ED8(FieldSystem *fsys, u16 *varPointer) {
+    u16 **didHeadbuttStartBattle = AllocFromHeapAtEnd(HEAP_ID_FIELD, sizeof(didHeadbuttStartBattle));
+    *didHeadbuttStartBattle = varPointer;
+    *varPointer = FALSE;
+    TaskManager_Call(fsys->taskman, ov01_02204EFC, didHeadbuttStartBattle);
+}
+
+static BOOL ov01_02204EFC(TaskManager *taskManager) {
+    UnkStruct_02204EFC *r4;
+    FieldSystem *fsys = TaskManager_GetFieldSystem(taskManager);
+    u16 **didHeadbuttStartBattle = TaskManager_GetEnv(taskManager);
```

Comment:
this is likely a struct pointer not a u16 **

Reply:
Do we really need to use a struct here if all it contains is just a `u16*`?

Reply:
The field environment variables will contain more than just the u16, that's just the only variable being referenced here

Reply:
Turns out that's all it contains.